### PR TITLE
[MM-47468] Fix store metrics initialization

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
There was a regression in database metrics due to a bug in the server initialization. We need to initialize metrics implementation before database gets initialized. Did some re-arrangement and added a test for this.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-47468

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/platform/metrics.go [hunk 1]
```diff
@@ -32,20 +32,18 @@ type platformMetrics struct {
...
-
-	return ps.metrics.metricsImpl
+	return ps.metricsIFace
 }
 
 // resetMetrics resets the metrics server. Clears the metrics if the metrics are disabled by the config.
-func (ps *PlatformService) resetMetrics(metricsImpl einterfaces.MetricsInterface, cfgFn func() *model.Config) error {
-	if !*cfgFn().MetricsSettings.Enable {
+func (ps *PlatformService) resetMetrics() error {
+	fmt.Println("resetMetrics")
```
Comment (agnivade) [inline app/platform/metrics.go:45]: Let's remove this

File: app/platform/metrics.go [hunk 2]
```diff
@@ -32,20 +32,18 @@ type platformMetrics struct {
 
 	metricsImpl einterfaces.MetricsInterface
 
-	cfgFn func() *model.Config
+	cfgFn      func() *model.Config
+	listenAddr string
 }
 
 func (ps *PlatformService) metricsImpl() einterfaces.MetricsInterface {
```
Comment (agnivade) [inline app/platform/metrics.go:39]: Are we using this method anywhere now? It seems we are directly accessing `ps.metricsIface`

Comment (isacikgoz) [reply app/platform/metrics.go:39]: Yeah, makes sense. I rushed to push the fix w/o properly reviewing it 😓

**Review by agnivade — APPROVED** [review]

**Review by agarciamontoro — APPROVED** [review]
This part of the code is indeed complex. Thanks for the fix, Ibrahim :)

**Comment by isacikgoz** [comment]
@saturninoabril do you spot any problems on the `e2e` tests for this PR? On the first attempt; 3 out of 20 jobs were failed but when I re-run them, all of them passed. Are we good to merge?

**Comment by saturninoabril** [comment]
Looking good so far. I'm just running the HA test and then will let you know. Thanks @isacikgoz!

**Review by saturninoabril — APPROVED** [review]
Thanks @isacikgoz, LGTM.

**Pull request merged**